### DISCUSSION TITLE
notebooks/fds/3

### DISCUSSION THREAD

**Discussion by giscus** [NoughtQ/notebook#9] [Announcements]
# notebooks/fds/3

https://note.noughtq.top/notebooks/fds/3

**Comment by RalphFH** [comment]
q->front = (PtrToQueue)malloc(sizeof(struct QueueNode));
q->rear = (PtrToQueue)malloc(sizeof(struct QueueNode));
q->front = q->rear;
这里队列创建的时候多申请了一块空间

**Reply by NoughtQ** [reply]
我记得当时做题目的时候，如果只分配一个空间，当我改变q->rear的时候，q->front也会跟着一起动，所以我就这么写了，我还是觉得应该要这么分配空间。当然，也有可能是我指针这块没有学好，以上只是我的个人想法
